flexget/plugins/list/radarr_list.py

```diff
@@ -0,0 +1,378 @@
+                url='',
+                radarr_id=movie['id'],
+                movie_name=movie['title'],
+                movie_year=movie['year']
+            )
+
+            # There seem to be a bug in the Radarr API because sometimes
+            # the imdbId is omitted in the response. So we can't be sure
+            # it's there
+            if 'imdbId' in movie:
+                entry['imdb_id'] = movie['imdbId']
+
+            if 'tmdbId' in movie:
+                entry['tmdb_id'] = movie['tmdbId']
+
+            if len(quality_requirements) > 0:
+                entry['quality_req'] = [str(quality_req) for quality_req in quality_requirements]
+
+            entries.append(entry)
+
+        return entries
+
+    def __lookup_movie(self, title=None, imdb_id=None, tmdb_id=None):
+        """
+        Uses Radarr's API to lookup a movie, prioritizing IMDB/TMDB
+        ids and as a last resort search for the title
+        """
+
+        # If the entry has a IMDB id, use that for lookup
+        if imdb_id:
+            try:
+                result = self.service.lookup_by_imdb(imdb_id)
+                # lookup_by_imdb returns an empty dictionary in case no match is found
+                if result:
+                    return result
+            except RadarrRequestError as ex:
+                log.error('The IMDB lookup raised exception: %s', ex)
+
+        # If the entry has a TMDB id, use that for lookup
+        if tmdb_id:
+            try:
+                result = self.service.lookup_by_tmdb(tmdb_id)
+                # lookup_by_tmdb returns an empty dictionary in case no match is found
+                if result:
+                    return result
+            except RadarrRequestError as ex:
+                log.error('The TMDB lookup raised exception: %s', ex)
+
+        # Could not lookup by id. Try to use the title.
+        # However, we can only accept any results if it's
+        # one item, otherwise we don't know which to select.
+        if title:
+            try:
+                results = self.service.lookup_by_term(title)
+                if len(results) > 1:
+                    log.debug(
+                        'The lookup for \'%s\' returned %d results. Using the first result \'%s\'.',
+                        title,
+                        len(results),
+                        results[0]['title']
+                    )
+                    return results[0]
+            except RadarrRequestError as ex:
+                log.error('The search term lookup raised exception: %s', ex)
+
+        return None
+
+class RadarrList(object):
+    """ List plugin for Radarr that also works as an input plugin """
+
+    schema = {
+        'type': 'object',
+        'properties': {
+            'base_url': {'type': 'string'},
+            'port': {'type': 'number', 'default': 80},
+            'api_key': {'type': 'string'},
+            'only_monitored': {'type': 'boolean', 'default': True},
```

Comment:
Good enhancements here, that would need to be ported to sonarr_list also, would be:

- monitored, type string, values yes, no, any, default any (replacing only_monitored)
- profile, type string, no default (default add profile)

It never made sense to me to handle the quality profiles as an input plugin but I guess somebody has a use for that.